action.yaml

```diff
@@ -250,26 +274,89 @@ runs:
           echo "IMAGE_DEPLOY_ONLY=false" >> $GITHUB_OUTPUT
 
           # set action
-          ACTION=deploy
+          ACTION=${{ inputs.action }}
         fi
 
         if [[ $ACTION == nothing ]]; then
           echo ERROR: you specified an improper action input. Action must be deploy, deploy-deployment-preview, create-deployment-preview, or delete-deployment-preview.
           exit 1 # terminate and indicate error
         fi
+        echo ::endgroup::
       shell: bash
       id: deployment-preview
     - name: Determine if DAG Deploy is enabled
       run: |
+
+        echo ::group::Determine if DAG Deploy is enabled
         if [[ ${{steps.deployment-preview.outputs.SKIP_DEPLOY}} == false && ${{steps.deployment-preview.outputs.IMAGE_DEPLOY_ONLY}} == false ]]; then
           echo "DAG_DEPLOY_ENABLED=$(astro deployment inspect ${{steps.deployment-preview.outputs.FINAL_DEPLOYMENT_ID}} --clean-output --key configuration.dag_deploy_enabled)" >> $GITHUB_OUTPUT
         else
           echo "DAG_DEPLOY_ENABLED=false" >> $GITHUB_OUTPUT
         fi
+        echo ::endgroup::
       shell: bash
       id: dag-deploy-enabled
+    - name: Get DBT Deploy Options
+      run: |
+
+        echo ::group::Get DBT Deploy Options
+        if [[ "${{ inputs.action }}" != "dbt-deploy" && "${{ inputs.action }}" != "dbt-deploy-deployment-preview" ]]; then
+          echo "skipping the steps since the action is not dbt-deploy or dbt-deploy-deployment-preview"
+          echo "DBT_DEPLOY=0" >> $GITHUB_OUTPUT
+          exit 0
+        fi
+        cd ${{ inputs.root-folder }}
+        branch=$(echo "${GITHUB_REF#refs/heads/}")
+        echo "Branch pushed to: $branch"
+        git fetch origin $branch
+        files=$(git diff --name-only ${{ github.event.before }} ${{ github.event.after }})
+        echo "Files changed: $files"
+        dbt_deploy=0
+
+        for file in $files; do
+          if [[ $file == *"${{ inputs.root-folder }}"* ]]; then
+            echo $file is part of dbt folder, so would be triggering a dbt deploy also
+            dbt_deploy=1
+          fi
+        done
+
+        # validate that the DAG Deploy is enabled for the Deployment
+        if [[ ${{steps.dag-deploy-enabled.outputs.DAG_DEPLOY_ENABLED}} == false && dbt_deploy == 1 ]]; then
+          echo ERROR: "Enable DAG Deploy for the Deployment in order to do a DBT Deploy"
+          exit 1
+        fi
+
+        # validate that mount path is provided
```

Comment:
Do we want the mount path to be required here? I think we can rely on the default mount path of the dbt Deploy (i.e. the directory name)? 

Also noticed the mount-path input above not required (https://github.com/astronomer/deploy-action/blob/c7900c85152ebbcd364fd354260e7194e5536929/action.yaml#L85) so if we do make it required I assume we should change

Reply:
cc @jeremybeard